spring-security-basic-auth/src/main/java/org/baeldung/spring/SecSecurityConfigJava.java

```diff
@@ -0,0 +1,40 @@
+package org.baeldung.spring;
+
+import org.baeldung.security.basic.MyBasicAuthenticationEntryPoint;
+import org.baeldung.security.filter.CustomFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@ComponentScan("org.baeldung.security")
+public class SecSecurityConfigJava extends WebSecurityConfigurerAdapter {
+    @Autowired
+    private MyBasicAuthenticationEntryPoint authenticationEntryPoint;
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+        auth
+          .inMemoryAuthentication()
+          .withUser("user1").password("user1Pass").authorities("ROLE_USER");
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .antMatchers("/securityNone").permitAll()
```

Comment:
This path was present in the XML config, so I just did the same in the Java config. I can see from the application that this path we can enter manually to see that it is not under security while homepage is under security.